src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList(int initCapacity) {
+        if (initCapacity <= 0) {
+            throw new IllegalArgumentException("initCapacity can`t be less than 1");
+        }
+        elementData = new Object[initCapacity];
+    }
+
+    public ArrayList() {
+        this(DEFAULT_CAPACITY);
+    }
+
     @Override
     public void add(T value) {
-
+        increaseDataArrayIfFull();
```

Comment:
Won't calling the function load the stack a little more each time, maybe it's better to do an additional check?

Reply:
This function is not used recursively, it used just a few times. I don't think that calling it has much effect on the overall performance of the application.